jobs/submanPipeline.groovy

```diff
@@ -1,131 +1,131 @@
 import hudson.plugins.ircbot.v2.IRCConnectionProvider
-import groovy.json.StringEscapeUtils
+import jenkins.model.Jenkins
 
-String REPO = 'https://github.com/candlepin/subscription-manager'
-String GITHUB_CREDENTIALS_ID = 'github-api-token-as-username-password'
-String GITHUB_ACCOUNT = 'candlepin'
-String GITHUB_REPO = 'subscription-manager'
-String GITHUB_COMMIT = ghprbActualCommit ?: sha1
-String PENDING_MESSAGE = 'Build has been scheduled.'
-String SUCCESS_MESSAGE = 'Test(s) passed.'
-String FAILURE_MESSAGE = 'Test(s) failed.'
+REPO = 'https://github.com/candlepin/subscription-manager'
+GITHUB_CREDENTIALS_ID = 'github-api-token-as-username-password'
+GITHUB_ACCOUNT = 'candlepin'
+GITHUB_REPO = 'subscription-manager'
+GITHUB_COMMIT = ghprbActualCommit ?: sha1
+PENDING_MESSAGE = 'Build has been scheduled.'
+SUCCESS_MESSAGE = 'Test(s) passed.'
+FAILURE_MESSAGE = 'Test(s) failed.'
 
-def sendIrcNotification = { channel, text ->
+STATUS_MESSAGE_MAP = [
+    'PENDING': PENDING_MESSAGE,
+    'SUCCESS': SUCCESS_MESSAGE,
+    'FAILURE': FAILURE_MESSAGE,
+]
+
+def sendIrcNotification(channel, text) {
     try {
         IRCConnectionProvider.getInstance().currentConnection().send(channel, text)
     } catch(e) {
         echo "Unable to send IRC notification to ${channel}: ${e}"
     }
 }
 
+def githubStatus(Map args) {
+    description = args.get('description') ?: STATUS_MESSAGE_MAP[args.status]
+    try {
+        githubNotify(account: GITHUB_ACCOUNT, repo: GITHUB_REPO, credentialsId: GITHUB_CREDENTIALS_ID, sha: GITHUB_COMMIT,
+                status: args.status, description: description, context: args.context, targetUrl: args.targetUrl)
+    } catch (e) {
+        echo "Unable to update GitHub status (commit: ${GITHUB_COMMIT}, status: ${args.status}, context: ${args.context})"
+    }
+}
+
+def buildWithNotifications(Map args) {
+    boolean enabled = Jenkins.instance.getItem(args.job, currentBuild.rawBuild.parent, Item.class).buildable
```

Comment:
Perhaps surround this in a try catch. I've recently seen with the candlepin pipeline somewhat cryptic failure messages that arise due to changes to the job name (Changed one job from "Candlepin Performance" to "CandlepinPerformance").

Non-nacking, but please consider!